scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/FTPConnector.java

```diff
@@ -0,0 +1,381 @@
+import org.ow2.proactive.scheduler.common.task.executable.JavaExecutable;
+
+
+/**
+ * Import and export file(s) using FTPConnector server.
+ * This task can be launched with parameters.
+ *
+ * @author The ProActive Team
+ */
+public class FTPConnector extends JavaExecutable {
+
+    private static final String DEFAULT_FTP_HOSTNAME = "localhost";
+
+    private static final int DEFAULT_FTP_PORT = 21;
+
+    private String ftpHostname = DEFAULT_FTP_HOSTNAME;
+
+    private int ftpPort = DEFAULT_FTP_PORT;
+
+    private boolean ftpExtractArchive;
+
+    private String ftpLocalFilePath = null;
+
+    private String ftpRemoteFilePath = null;
+
+    private String ftpMode = null;
+
+    private String ftpUsername = null;
+
+    private String ftpPassword = null;
+
+    /**
+     * @see JavaExecutable#init(Map)
+     */
+    @Override
+    public void init(Map<String, Serializable> args) throws Exception {
+        if (args.containsKey("ftpHostname")) {
+            ftpHostname = args.get("ftpHostname").toString();
+        }
+        if (args.containsKey("ftpPort")) {
+            try {
+                ftpPort = Integer.parseInt(args.get("ftpPort").toString());
+            } catch (NumberFormatException e) {
+            }
+        }
+        if (args.containsKey("ftpExtractArchive")) {
+            ftpExtractArchive = Boolean.parseBoolean(args.get("ftpExtractArchive").toString());
+        }
+        if (args.containsKey("ftpLocalFilePath")) {
+            ftpLocalFilePath = args.get("ftpLocalFilePath") == null ? "" : args.get("ftpLocalFilePath").toString();
+        }
+        if (args.containsKey("ftpRemoteFilePath")) {
+            ftpRemoteFilePath = args.get("ftpRemoteFilePath").toString();
+        }
+        if (args.containsKey("ftpMode")) {
+            ftpMode = args.get("ftpMode").toString();
+        }
+        ftpUsername = getThirdPartyCredential("FTP_USERNAME");
+        ftpPassword = getThirdPartyCredential("FTP_PASSWORD");
+        if (ftpUsername == null || ftpPassword == null) {
+            throw new Exception("You first need to add your ftp username and password (FTP_USERNAME, FTP_PASSWORD) to the third party credentials");
+        }
+    }
+
+    /**
+     * @see JavaExecutable#execute(TaskResult[])
+     */
+    @Override
+    public Serializable execute(TaskResult... results) throws Exception {
+
+        FTPClient ftpClient = new FTPClient();
+        try {
+            // connect and login to the server
+            ftpClient.connect(ftpHostname, ftpPort);
+            int reply = ftpClient.getReplyCode();
+            if (!FTPReply.isPositiveCompletion(reply)) {
+                ftpClient.disconnect();
+                throw new Exception("Exception in connecting to FTP Server");
+            }
+            ftpClient.login(ftpUsername, ftpPassword);
+
+            // use local passive mode to pass firewall
+            ftpClient.enterLocalPassiveMode();
+
+            System.out.println("Connected");
```

Comment:
remove all System.out.println and use getOut().println instead.

similarly System.err.println should be replaced by getErr().println